frontend/gateway/grpcclient/client.go

```diff
@@ -386,31 +386,27 @@ func (c *grpcClient) Solve(ctx context.Context, creq client.SolveRequest) (res *
 	if creq.Evaluate {
 		if c.caps.Supports(pb.CapGatewayEvaluateSolve) == nil {
 			req.Evaluate = creq.Evaluate
-		} else {
-			// If evaluate is not supported, fallback to running Stat(".") in order to
-			// trigger an evaluation of the result.
+		} else if c.caps.Supports(pb.CapGatewayEvaluate) == nil {
```

Comment:
Should we not check `Evaluate` cap first?

Reply:
Removing this block of code :smile: